crates/service_utils/src/service/types.rs

```diff
@@ -228,3 +240,31 @@ impl FromRequest for CustomHeaders {
         ready(Ok(val))
     }
 }
+
+pub struct CustomResp<T> {
+    pub body: T,
+    pub headers: HashMap<String, String>,
+}
+impl<T: Serialize> Responder for CustomResp<T> {
+    type Body = EitherBody<String>;
```

Comment:
This should be 
```suggestion
    type Body = EitherBody<T>;
```
right?